libnavigation-ui/src/main/java/com/mapbox/navigation/ui/MapOfflineManager.java

```diff
@@ -53,6 +53,7 @@ public void onRouteProgressChanged(@NotNull RouteProgress routeProgress) {
     Geometry currentRouteGeometry = routeProgress.getRouteGeometryWithBuffer();
     if (previousRouteGeometry == null || !previousRouteGeometry.equals(currentRouteGeometry)) {
       previousRouteGeometry = currentRouteGeometry;
+      // TODO Method invocation 'requestUuid' may produce 'NullPointerException'
```

Comment:
cc @abhishek1508 @JunDai @langsmith @cafesilencio

Reply:
what would happen if requestUuid was `""`?

Reply:
Problem here is that `RouteOptions` is `@Nullable` 😬

Reply:
what should happen? a no op?

Reply:
@abhishek1508 @JunDai @langsmith @cafesilencio Could you 👀 when you have a chance?

Reply:
as doc mentioned, the **requestUuid** will not be `NULL`.
```java
  /**
   * A universally unique identifier (UUID) for identifying and executing a similar specific route
   * in the future. <tt>MapboxDirections</tt> always waits for the response object which ensures
   * this value will never be null.
   *
   * @return a string containing the request UUID
   * @since 3.0.0
   */
  @SerializedName("uuid")
  @Nullable
  public abstract String requestUuid();
```

For the **routeOptions**, it's annotated as `Nullable` but I doubt it? It worths to check with DirectionAPI team.
When it's null, we are not able to download and should just return I think.

Reply:
LMAO

```java
<tt>MapboxDirections</tt> always waits for the response object which ensures
   * this value will never be null.
```

but it's `@Nullable` it seems we need to fix in https://github.com/mapbox/mapbox-java 😅

Reply:
Could you cut a ticket in https://github.com/mapbox/mapbox-java/issues/new when you have a chance @JunDai 🙏?

Reply:
> Could you cut a ticket in https://github.com/mapbox/mapbox-java/issues/new when you have a chance @JunDai 🙏?

sure thing